File: store_data_at_delete-setting-case-java/src/main/java/com/example/demo/memo/MemoGenerationRequest.java
package com.example.demo.memo;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Size;

import java.time.LocalDateTime;

public record MemoGenerationRequest(
        @Schema(requiredMode = Schema.RequiredMode.REQUIRED, description = "메모 제목", example = "Lorem Ipsum")
        @Size(min = 1, max = 100)
        String title,

        @Schema(requiredMode = Schema.RequiredMode.REQUIRED, description = "메모 내용", example = "Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit...")
        @Size(min = 1, max = 16_777_215)  // ~ MEDIUMTEXT (16MB)
        String content
) {
    public MemoEntity toEntity() {
        var now = LocalDateTime.now();
        return new MemoEntity(null, title, content, now, now);
    }
}

File: store_data_at_delete-setting-case/src/main/kotlin/com/example/demo/memo/MemoGenerationRequest.kt
package com.example.demo.memo

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.Size
import java.time.LocalDateTime

@Schema(description = "메모 생성(신규/수정) 요청")
data class MemoGenerationRequest(
    @field:Schema(required = true, description = "메모 제목", example = "Lorem Ipsum")
    @field:Size(min = 1, max = 100)
    val title: String,
    @field:Schema(required = true, description = "메모 내용", example = "Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit...")
    @field:Size(min = 1, max = 16_777_215)  // ~ MEDIUMTEXT (16MB)
    val content: String,
) {
    fun toEntity(): MemoEntity {
        val now: LocalDateTime = LocalDateTime.now()
        return MemoEntity(
            title = title,
            content = content,
            createdAt = now,
            updatedAt = now,
        )
    }
}
